### DISCUSSION TITLE
keyball44_Ball_Side 右手利用時「.>」キーが反応せず。ダイオード位置の確認と keyball シリーズのダイオード配置について

### DISCUSSION THREAD

**Discussion by tyuu-com** [Yowkees/keyball#427] [Q&A]
Keyball44 を購入。組み立てております。
掲題の通り「.>」裏面で言う LED20 のキー（緑枠）が反応しませんでした。

質問.1
疑わしきダイオードを教えてください。
![IMG_0660](https://github.com/Yowkees/keyball/assets/152047617/b16db415-9b48-4364-bc4d-9141b7cc259d)
テスターを当て導通確認できたスイッチに赤矢印を記載しました。
掲題のキーが対応するのは  LED20 の上段ダイオードかと思われますが、間違いないでしょうか？

テスターで確認しているのであれば、質問する必要ないのでは？と思われるかもしれません。
「割り当てのない、または重複するダイオード」が見受けられ、配線に自信がなくなり質問させていただきました。
「割り当てのない、または重複するダイオード」は「質問.2」で質問させてください。

質問.2
「割り当てのない、または重複するダイオード」がありました。
これら半田づけミスの疑いありますでしょうか？

「割り当てのないダイオード」
　・LED19 の上にあるダイオード
　・LED20 の上にあるダイオード
　　近くのキーと導通確認できず、そもそもでキーとしては充足されており、用途がわかりませんでした。
「重複するダイオード」
　・LED15 の 2 つ上にあるダイオード
　　「P」キーと「P 右（通常英字キーボード「[{」）」両方と導通が確認できました。
　　問題なく利用できているように思ったのでうが、半田ミスでしょうか？

質問.3
　「keyball シリーズのダイオード配置」について、どれがどこに対応しているか、解説しているドキュメントはございませんでしょうか？
　今回、該当のダイオードがどれか Discussion 中に、先人たちが質問されてないか確認したのですが、
　都度、個別にご対応いただいている状況に見えました。
　今後 keyball61 の購入を検討しており、同様の半田ミスに悩みそうだったので、事前にダイオード配置と対応が確認できると心強いです。
　keyball61 の購入検討理由は keyball44 のキーレイヤーが「0...3」の 4 枚で足りなかったからです。。。
　自身でファームウェアをコンパイルするか、どうしようか悩んでおります。。。（が、これは質問ではございません。。。）

以上

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

質問１
おっしゃる通り、下記画像のオレンジ丸で示したダイオードが該当します。
ただ、キーが効かない原因の多くはキーソケット自体の半田付け不良の可能性のほうが高いです。
キーソケット端子を横から覗いて、基板パッドと端子の間に半田がきちんと流れ込んでいるかご確認ください。

![KB44R_diodefix](https://github.com/Yowkees/keyball/assets/84209772/d7efd5af-7267-4123-b38f-178a59891d0b)

質問２
上記画像の緑・黄色・青線で結んだ個所が対応しています。
LED17近くの上側のダイオードは左右判定に使用しているダイオードで、キーソケットには接続されていないダイオードになります。

質問３
下記リンクにKeyball44の回路図と基板データを公開しており、KiCADというフリーソフトで表示できますが、確かにキーソケットとダイオードの対応図みたいなものは公開するべきかもと感じました。
ぜひ順次準備したいと思います。
https://github.com/Yowkees/keyball/tree/main/keyball44/design_data

以上お手数ですが、半田付けのご確認よろしくお願いいたします。

**Reply by tyuu-com** [reply]
ご回答、ありがとうございました。

質問1 
結論からお伝えすると、無事にキー入力ができるようになりました。
キーソケットのはんだ付けで回復しました。

質問2 の回答納得しました。
質問3 の対応図はご検討ください。

ありがとうございました。

**Comment by Yowkees** [comment]
無事に入力できるようになり良かったです。安心しました。
対応図はぜひ準備させていただきます。

ぜひ末永くご愛用いただければ幸いです。
